modules/minio/minio.go

```diff
@@ -0,0 +1,97 @@
+package minio
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	defaultUser     = "minioadmin"
+	defaultPassword = "minioadmin"
+	defaultImage    = "docker.io/minio/minio:RELEASE.2024-01-16T16-07-38Z"
+)
+
+// MinioContainer represents the Minio container type used in the module
+type MinioContainer struct {
+	testcontainers.Container
+	Username string
+	Password string
+}
+
+// WithUsername sets the initial username to be created when the container starts
+// It is used in conjunction with WithPassword to set a user and its password.
+// It will create the specified user. It must not be empty or undefined.
+func WithUsername(username string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		if username == "" {
+			username = defaultUser
+		}
+		req.Env["MINIO_ROOT_USER"] = username
+	}
+}
+
+// WithPassword sets the initial password of the user to be created when the container starts
+// It is required for you to use the Minio image. It must not be empty or undefined.
+// This environment variable sets the root user password for Minio.
+func WithPassword(password string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) {
+		if password == "" {
+			password = defaultPassword
+		}
+		req.Env["MINIO_ROOT_PASSWORD"] = password
+	}
+}
+
+// ConnectionString returns the connection string for the minio container, using the default 9000 port, and
+// obtaining the host and exposed port from the container.
+func (c *MinioContainer) ConnectionString(ctx context.Context) (string, error) {
+	host, err := c.Host(ctx)
+	if err != nil {
+		return "", err
+	}
+	port, err := c.MappedPort(ctx, "9000/tcp")
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("%s:%s", host, port.Port()), nil
+}
+
+// RunContainer creates an instance of the Minio container type
+func RunContainer(ctx context.Context, opts ...testcontainers.ContainerCustomizer) (*MinioContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        defaultImage,
+		ExposedPorts: []string{"9000/tcp"},
+		WaitingFor:   wait.ForListeningPort("9000").WithStartupTimeout(time.Minute * 2),
```

Comment:
an accurate implementation should be implemented using
```suggestion
		WaitingFor:   wait.ForHTTP("/minio/health/live").WithPort("8080"),
```

Reply:
Thanks for the heads up. Also is the default version that I used ok or would you prefer an earlier build?